curriculum/challenges/english/25-front-end-development/lab-nth-fibonacci-number-generator/66d9af3897e7d75a895b72c2.md

```diff
@@ -0,0 +1,116 @@
+---
+id: 66d9af3897e7d75a895b72c2
+title: Build the nth Fibonacci number generator
+challengeType: 14
+dashedName: lab-nth-fibonacci-number-generator
+---
+
+# --description--
+
+In this lab you will use a dynamic programming approach to compute and return the `n`-th number from the Fibonacci sequence, in which each number is the sum of the two preceding numbers.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should define an array named `sequence` in the global scope. It should contain the number `0` and `1`.
+1. You should create a function named `fibonacci`.
+1. The `fibonacci` function should take a single parameter, which is a positive integer.
+1. If the input is not a positive integer, the function should return `"Invalid input"`.
+1. A call to `fibonacci(n)` should use a dynamic programming approach to compute and return the `n`-th number from the Fibonacci sequence.
+1. Each computed number at the position `n` in the Fibonacci sequence should be stored at the index `n - 1` of the `sequence` array.
+1. If the input number requests an already computed number, no new computations should happen and the number should be retrieved from the `sequence` array.
+
+# --hints--
+
+You should have a `sequence` array.
+
+```js
+assert.isArray(sequence);
+```
+
+The `sequence` array should contain `0` and `1` at definition.
+
+```js
+assert.match(__helpers.removeJSComments(code), /(let|const)\s+sequence\s*=\s*\[\s*0\s*,\s*1\s*\]\s*;?/)
+```
+
+You should create a `fibonacci` function.
+
+```js
+assert.isFunction(fibonacci);
+```
+
+If the `fibonacci` function receives an argument that is not a positive integer it should return `"Invalid input"`.
+
+```js
+assert.strictEqual(fibonacci(3.5), "Invalid input")
+assert.strictEqual(fibonacci(0), "Invalid input")
```

Comment:
Zero is not a positive nor a negative number. We should remove this or use a different input to test.

Reply:
I was about to say the same, but the user story is accurate because it says "If the input is not a positive integer" so everything else is excluded. So I think this can be kept as is.
One thing that I would clarify is that tests are checking that the type is not different from "number".

Reply:
I sort of agree with you. While in fact it states "if the input is not a positive integer.". We should make clear to the camper that the input could be of any kind.

This will otherwise cause too many issues on the forum.

Reply:
I can remove the string test, but I see no reason to remove the 0

Reply:
Let me reiterate, I think it's fine, but the user has too little context to go on. Most of the time when completing coding challenges the user is expected to exclude any negative number (If the user story is phrased like this).

 We could write an extra hint that checks invalid values?

Like: The fibonacci function should not accept `3.5` as a valid input.

and: The fibonacci function should not accept `0` as a valid input.

Reply:
divide it into multiple tests? shure